source/extensions/filters/http/local_ratelimit/local_ratelimit.h

```diff
@@ -70,9 +70,16 @@ class PerConnectionRateLimiter : public StreamInfo::FilterState::Object {
 class FilterConfig : public Router::RouteSpecificFilterConfig {
 public:
   FilterConfig(const envoy::extensions::filters::http::local_ratelimit::v3::LocalRateLimit& config,
-               const LocalInfo::LocalInfo& local_info, Event::Dispatcher& dispatcher,
+               const LocalInfo::LocalInfo& local_info, Event::Dispatcher& main_dispatcher,
                Stats::Scope& scope, Runtime::Loader& runtime, bool per_route = false);
-  ~FilterConfig() override = default;
+  ~FilterConfig() override {
+    if (!main_dispatcher_.isThreadSafe()) {
```

Comment:
I would probably just skip this check and always post for consistency. I don't think it needs to be optimized. Can you also add more comments on why this is needed?